### PR TITLE
Task aicore 553 index baf op

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

**Comment by andreinechaev** [comment]
only last commit needs to be reviewed - the rest is in #487

File: nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/services/SimilarServiceComponent.java [hunk 1]
```diff
@@ -84,6 +113,91 @@ public String getOperationID() {
...
+                                   .map(BulkCodecs.getStatusCodec()::decode)
+                                   .filter(status -> INDEX_ACTION_NAME.equals(status.getAction()))
+                                   .anyMatch(status -> status.getState().equals(BulkStatus.State.RUNNING)
+                                           || status.getState().equals(BulkStatus.State.SCHEDULED));
+        if (alreadyRunning) {
+            throw new ConcurrentUpdateException("Only one deduplication index action is allowed at a time");
+        }
+
+        if (!reindex) {
+            query += " AND ecm:mixinType != " + NXQL.escapeString(DEDUPLICATION_FACET);
```
Comment (vpasquier) [inline nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/services/SimilarServiceComponent.java:131]: Should we not have this check on the query for the facet filter for all use cases?

**Review by vpasquier — APPROVED** [review]
Just a question but all good!

File: nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/services/SimilarServiceComponent.java (hunk 1 above)
Comment (andreinechaev) [reply nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/services/SimilarServiceComponent.java:131]: we can indeed check if such expression already takes place and apply it optionally. If initial query wasn't properly structured and contains this condition - it's a considered as a human error IMO

**Pull request merged**